### PR TITLE
Josephus Problem is solved

### PR DESCRIPTION
**Pull request by Devchawla2608**
## 🛠️ Issue 1318
#1318 

## ✔️ Check List (Check all the applicable boxes) 

- [ ] My code follows the code style of this project.
- [ ] This PR does not contain plagiarized content.
- [ ] The title of my pull request is a short description of the requested changes.

## 📷 Screenshot
image

### CODE REVIEW

**Review by Kumar-laxmi — COMMENTED** [review]
image

Does it mean your contribution doesn't follow any of these

**Review by Kumar-laxmi — CHANGES_REQUESTED** [review]
Your implementation in C, Java and Python is yet to be done

**Review by zugzwang03 — APPROVED** [review]

**Review by Arun9739 — APPROVED** [review]

**Review by 072arushi — APPROVED** [review]

File: C++/Maths/Josephus.cpp [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+// 2. Initialize the list with people numbered from 1 to n.
+// 3. Start with an iterator pointing to the first person.
+// 4. Repeat the elimination process until only one person remains:
+//     a. Move the iterator k-1 steps in a circular manner.
+//     b. Erase the person at the current position.
+//     c. Update the iterator to the next position after the elimination, wrapping around if necessary.
+// 5. Return the position of the last remaining person.
+
+int josephus(int n, int k) {
+    std::list<int> circle;
```
Comment (Kumar-laxmi) [inline C++/Maths/Josephus.cpp:16]: Why did you use ```std``` when you have already written ```using namespace std;``` at line 3

File: C++/Maths/Josephus.cpp [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    return circle.front();  // Return the position of the last remaining person
+}
+
+int main() {
+    int n;  // Number of people in the circle
+    int k;   // Elimination step size
+    cout<<"Enter n and k "<<endl;
+    cin>>n>>k;
+    int last_person = josephus(n, k);  // Solve the Josephus Problem
+    std::cout << "The last remaining person is at position: " << last_person << std::endl;
```
Comment (Kumar-laxmi) [inline C++/Maths/Josephus.cpp:45]: Why did you use ```std``` when you have already written ```using namespace std;``` at line 3

**Review by Kumar-laxmi — CHANGES_REQUESTED** [review]

File: C++/Maths/Josephus.cpp (hunk 1 above)
Comment (Devchawla2608) [reply C++/Maths/Josephus.cpp:16]: For list function. Sir did not know about the list function that belongs to the namespace so I mentioned

**Comment by Devchawla2608** [comment]
Sir all requested changes are done

**Review by Kumar-laxmi — APPROVED** [review]

**Pull request merged**